constants/src/main/java/org/jboss/pnc/enums/BuildStatus.java

```diff
@@ -55,51 +55,58 @@
     /**
      * Build rejected due to conflict with another build, or failed dependency build
      */
-    REJECTED,
+    REJECTED(false),
 
     /**
      * User cancelled the build
      */
-    CANCELLED,
+    CANCELLED(true),
 
     /**
      * A system error prevented the build from completing
      */
-    SYSTEM_ERROR,
+    SYSTEM_ERROR(false),
 
     /**
      * It is not known what the build status is at this time
      */
-    UNKNOWN,
+    NEW;
 
-    /**
-     * There have not been any builds of this configuration, essentially the same meaning
-     * as a null status
-     */
-    NONE;
+    // isFinal means it's in the final state, it could also have failed
+    private final boolean isFinal;
```

Comment:
Why do we need this boolean?

Reply:
To know if the build is in a final state. Aka if it's done (whether successful or not)

Reply:
Do we have any use-case when we need it?

Reply:
Yes: https://github.com/project-ncl/pnc/blob/master/build-coordinator/src/main/java/org/jboss/pnc/coordinator/notifications/buildTask/BuildStatusNotifications.java#L66 (this is using BuildCoordinatorStatus, so we need an equivalent for BuildStatus)